fennel/sources/test_sources.py

```diff
@@ -224,6 +225,114 @@ class UserInfoDataset:
 )
 
 
+def test_tier_selector_on_source():
```

Comment:
Worth adding a test where multiple sources/pipelines/extractors exist but none get selected in the end?